internal/services/compute/virtual_machine.go

```diff
@@ -285,7 +303,8 @@ func flattenVirtualMachineOSDisk(ctx context.Context, disksClient *compute.Disks
 	diffDiskSettings := make([]interface{}, 0)
 	if input.DiffDiskSettings != nil {
 		diffDiskSettings = append(diffDiskSettings, map[string]interface{}{
-			"option": string(input.DiffDiskSettings.Option),
+			"option":    string(input.DiffDiskSettings.Option),
+			"placement": string(input.DiffDiskSettings.Placement),
```

Comment:
this'll be nil/empty in the response for existing resources, so we'd need to default this?

Reply:
The original version has it set, but I've asked not to: https://github.com/hashicorp/terraform-provider-azurerm/pull/14847#discussion_r844908954 😅.

My understanding is the [original version](https://github.com/hashicorp/terraform-provider-azurerm/pull/14847/commits/141c83f13a06b11ad693f04a4bb9999c67a1f166#diff-2ef599aaaa2dc4bfdd95b39fbe225bc832397021568eb63013ef9d15c581e082R303) is trying to check a low likely case that the API didn't return the `placement`, then set it to its default value. But that is not always right. Imagine users have set the `placement` to `DiffDiskPlacementResourceDisk`, then if the API didn't return it, it will still ends up with a diff as the default value is `DiffDiskPlacementCacheDisk`. So I prefer just leave it as is.

Reply:
@tombuildsstuff @magodo thanks for sharing the thoughts. I took another look, this property is not introduced initially in service, and default value is `CacheDisk`, so when it's missing from the response, we could probably consider it as `CacheDisk`. If it's set to `ResourceDisk`, it should not be missing in the response.